### DISCUSSION TITLE
Validation types/profiles in biolink model

### DISCUSSION THREAD

**Discussion by kshefchek** [linkml/linkml#146] [Ideas]
[Moving from a discussion on slack]

In reviewing biolinkML @ShahimEssaid mentioned that fhir has support for validation profiles via [structured definitions](https://www.hl7.org/fhir/structuredefinition.html).  @hsolbrig we were curious your thoughts given your familiarity with both modelling languages.  Perhaps the conversion of biolinkML yaml to shex and jsonschema serve similar purposes.

**Comment by hsolbrig** [comment]
My take (fwiw) is that FHIR is a modeling language, and a pretty good one at that.  We are trying to fill a different set of use cases w/ the language formally known as BiolinkML, which is a) defining the semantics of models and b) using these definitions to do data instance transformations.  The bit that ends up confusing is, just as FHIR models StructureDefinition _using_ StructureDefinition, BiolinkML models BiolinkML using BiolinkML -- the model transformations are interesting, but the true value is (or will/should be) in model instance transformations.  The closest that FHIR comes to the latter is the FHIR Mapping Language....

See: https://github.com/biolink/biolinkml/issues/315 for more info on this approach

**Comment by cmungall** [comment]
Closing this. See https://linkml.io/linkml/data/validating-data.html for details of LinkML validation
